src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2536,306 +2536,318 @@ mgp_error mgp_graph_get_vertex_by_id(mgp_graph *graph, mgp_vertex_id id, mgp_mem
-          return property_ids_str;
-        },
+  return WrapExceptions([graph, memory, result]() {
+    const auto constraints_res = std::visit(
+        memgraph::utils::Overloaded{
+            [](memgraph::query::DbAccessor *impl) { return impl->ListAllConstraints().unique; },
+            [](memgraph::query::SubgraphDbAccessor *impl) { return impl->GetAccessor()->ListAllConstraints().unique; }},
         graph->impl);
 
-    mgp_list *label_properties_mgp_list = nullptr;
-    if (auto properties_mgp_list_err =
-            mgp_list_make_empty(properties_str.size() + 1, memory, &label_properties_mgp_list);
-        properties_mgp_list_err != mgp_error::MGP_ERROR_NO_ERROR) {
-      return properties_mgp_list_err;
-    }
+    if (auto err = mgp_list_make_empty(constraints_res.size(), memory, result); err != mgp_error::MGP_ERROR_NO_ERROR) {
+      throw std::logic_error("Cannot create list");
+    }
+
+    for (const auto &label_properties_pair : constraints_res) {
+      const std::string label_id_str =
+          std::visit([label_id = label_properties_pair.first](const auto *impl) { return impl->LabelToName(label_id); },
+                     graph->impl);
+      const std::vector<std::string> properties_str = std::visit(
+          [property_ids = label_properties_pair.second](const auto *impl) {
+            std::vector<std::string> property_ids_str;
+            property_ids_str.reserve(property_ids.size());
+            std::transform(property_ids.begin(), property_ids.end(), std::back_inserter(property_ids_str),
+                           [impl](const auto &property_id) { return impl->PropertyToName(property_id); });
+            return property_ids_str;
+          },
+          graph->impl);
 
-    mgp_value *mgp_value_label = nullptr;
-    if (auto err_label = mgp_value_make_string(label_id_str.c_str(), memory, &mgp_value_label);
-        err_label != mgp_error::MGP_ERROR_NO_ERROR) {
-      return err_label;
-    }
-    if (auto err_label_into_list = mgp_list_append_extend(label_properties_mgp_list, mgp_value_label);
-        err_label_into_list != mgp_error::MGP_ERROR_NO_ERROR) {
-      return err_label_into_list;
-    }
+      mgp_list *label_properties_mgp_list = nullptr;
+      if (auto properties_mgp_list_err =
+              mgp_list_make_empty(properties_str.size() + 1, memory, &label_properties_mgp_list);
+          properties_mgp_list_err != mgp_error::MGP_ERROR_NO_ERROR) {
+        throw std::logic_error("Cannot create list");
+      }
 
-    for (const std::string &property_str : properties_str) {
-      mgp_value *property_mgp_value = nullptr;
-      if (auto err_str = mgp_value_make_string(property_str.c_str(), memory, &property_mgp_value);
-          err_str != mgp_error::MGP_ERROR_NO_ERROR) {
-        return err_str;
+      mgp_value *mgp_value_label = nullptr;
+      if (auto err_label = mgp_value_make_string(label_id_str.c_str(), memory, &mgp_value_label);
```

Comment:
Same comment to this err_label about const-ness. If you agree, please apply this to all other err variables